### PR TITLE
Docs parity for find with JupyterLab.

### PR DESCRIPTION
**Pull request by ericsnekbytes**
This PR updates the find documentation to match changes in JupyterLab's. Questions that need to be answered:

- Can users disable windowed rendering in Notebook the same way they can in Lab, as suggested by @krassowski for Lab?

Original PRs for reference:

- Notebook find docs https://github.com/jupyter/notebook/pull/6905
- Lab find docs https://github.com/jupyterlab/jupyterlab/pull/14667

### CODE REVIEW

File: docs/source/notebook.md [hunk 1]
```diff
@@ -334,14 +334,17 @@ For the full list of available shortcuts, click {guilabel}`Help`,
 
 ## Searching
 
-Jupyter Notebook has an intelligent built-in search plugin for finding text within a
+Jupyter Notebook has a special built-in search plugin for finding text within a
```
Comment (jtpio) [inline docs/source/notebook.md:337]: Or maybe:

```suggestion
Jupyter Notebook has an advanced built-in search plugin for finding text within a
```

Comment (krassowski) [reply docs/source/notebook.md:337]: what about?

```suggestion
Jupyter Notebook has a dedicated built-in search plugin for finding text within a
```

Comment (ericsnekbytes) [reply docs/source/notebook.md]: I merged with "advanced".

**Comment by jtpio** [comment]
Cool, thanks for working on this :+1:

**Pull request merged**